### ISSUE TITLE
Change application name from default, via parameter -ApplicationName

### ISSUE THREAD

**Issue by pcgeek86** [Windos/PoshNotify#25]
On Windows 10, the notification toast has the application name showing as `Windows PowerShell`. Would it be possible to change the application name from "Windows PowerShell" to something else? Does the underlying API allow for this?

**Comment by Windos** [comment]
Yeah, it's possible but it's a whole thing. Basically it requires a shortcut in the start menu with an embedded AppId (full name for searching is AppUserModelId).

You can use `Get-StartApps` to see what's available on your system.

I've done some work trying to make custom AppId's more available to people:

https://king.geek.nz/2018/02/04/burnttoast-appid-installer/

https://king.geek.nz/2018/03/03/burnttoast-wix/

An alternative may be attribution, it'll still show PowerShell but you can put some text next to it.

![attribution gif](https://king.geek.nz/content/images/2018/10/Attrib.gif)

https://king.geek.nz/2018/10/05/crouton-9-attribution/

... I'm on phone, please forgive typos or bad markdown

**Comment by Windos** [comment]
Note in that gif, it shows BurntToast instead of PowerShell because I'm using my custom AppId.

**Comment by pcgeek86** [comment]
Oh yikes, that does sound involved, but a wrapper may be helpful to package up "fake" AppIds that can be used for this purpose. Appx management is thankfully straightforward with PowerShell.

Maybe use a dynamic parameter to detect Windows, and make the option only available on that platform?

**Comment by TylerLeonhardt** [comment]
> Maybe use a dynamic parameter to detect Windows, and make the option only available on that platform?

The consistency-obsessed guy in me is scared of this tbh... but the customer-obsessed guy in me hears you

**Comment by pcgeek86** [comment]
@TylerLeonhardt I'm not a big fan of dynamic parameters either, due to the well-known discoverability issues, but this scenario seems like the right fit for them. Maybe just make it a regular parameter and ignore it for non-Windows platforms?
